File: api/models/paste_create_info.go
package models

// Represents info used to create new pastes.
type PasteCreateInfo struct {
	Title     string            `json:"title" validate:"max=128"`
	Pasties   []PastyCreateInfo `json:"pasties" validate:"required,dive,min=1"`
	ExpiresIn ExpiresIn         `json:"expiresIn" validate:"omitempty,oneof=never 1h 2h 10h 1d 2d 1w 1m 1y"`
}

File: api/Models/PasteCreateInfo.cs
using System.ComponentModel.DataAnnotations;

namespace pastemyst.Models;

public class PasteCreateInfo
{
    [MaxLength(128)]
    public string Title { get; set; }
    
    [MinLength(1)]
    [Required]
    public List<PastyCreateInfo> Pasties { get; set; }

    public ExpiresIn ExpiresIn { get; set; } = ExpiresIn.Never;

    public bool Anonymous { get; set; } = false;

    public bool Private { get; set; } = false;

    public bool Pinned { get; set; } = false;
    
    public List<string> Tags { get; set; } = new();
}